src/Lykke.Quintessence.DomainServices/DefaultBlockchainService.cs

```diff
@@ -0,0 +1,229 @@
+using JetBrains.Annotations;
+using Lykke.Quintessence.Core.Blockchain;
+using Lykke.Quintessence.Core.Utils;
+using Lykke.Quintessence.Domain.Services.Strategies;
+using Lykke.Quintessence.Domain.Services.Utils;
+using Lykke.Quintessence.RpcClient;
+using Lykke.Quintessence.RpcClient.Exceptions;
+using Lykke.SettingsReader;
+using Newtonsoft.Json;
+
+
+namespace Lykke.Quintessence.Domain.Services
+{
+    [UsedImplicitly]
+    public class DefaultBlockchainService : IBlockchainService
+    {
+        private readonly SettingManager<int> _confirmationLevel;
+        private readonly IDetectContractStrategy _detectContractStrategy;
+        private readonly IEthApiClient _ethApiClient;
+        private readonly SettingManager<string, (BigInteger, BigInteger)> _gasPriceRange;
+        private readonly IGetTransactionReceiptsStrategy _getTransactionReceiptsStrategy;
+        private readonly ITryGetTransactionErrorStrategy _tryGetTransactionErrorStrategy;
+
+        
+        public DefaultBlockchainService(
+            IEthApiClient ethApiClient,
+            IDetectContractStrategy detectContractStrategy,
+            IGetTransactionReceiptsStrategy getTransactionReceiptsStrategy,
+            ITryGetTransactionErrorStrategy tryGetTransactionErrorStrategy,
+            Settings settings)
+        {
+            _ethApiClient = ethApiClient;
+            _detectContractStrategy = detectContractStrategy;
+            _getTransactionReceiptsStrategy = getTransactionReceiptsStrategy;
+            _tryGetTransactionErrorStrategy = tryGetTransactionErrorStrategy;
+            
+            _confirmationLevel = new SettingManager<int>
+            (
+                settings.ConfirmationLevel,
+                TimeSpan.FromMinutes(1)
+            );
+            
+            _gasPriceRange = new SettingManager<string, (BigInteger, BigInteger)>
+            (
+                settings.GasPriceRange,
+                ParseGasPriceRange,
+                TimeSpan.FromMinutes(1)
+            );
+        }
+
+        public virtual async Task<string> BroadcastTransactionAsync(
+            string signedTxData)
+        {
+            var serializedTransaction = signedTxData.HexToUTF8String();
+            var transaction = JsonConvert.DeserializeObject<DefaultRawTransaction>(serializedTransaction);
+
+            if (await _ethApiClient.GetTransactionAsync(transaction.Hash) != null)
+            {
+                return transaction.Hash;
+            }
+            
+            await _ethApiClient.SendRawTransactionAsync(transaction.Data);
+                
+            for (var i = 0; i < 10; i++)
+            {
+                if (await _ethApiClient.GetTransactionAsync(transaction.Hash) != null)
+                {
+                    return transaction.Hash;
+                }
+                else
+                {
+                    await Task.Delay(1000);
+                }
+            }
+                
+            throw new Exception
```

Comment:
Is it final state? Is it possible that tx not appears in mempool within 10 seconds but affects blockchain in any manner later?

Reply:
In this case API will return successful result on retry